### PR TITLE
664: Add `DataGrouped` class

### LINKED ISSUE #664: Add `DataGrouped` class
See https://github.com/Roche/crmPack/blob/main/design/single_combo_grouped.Rmd#L24.

To do:

- [x] Add class
    - [x] doc
	- [x] with validation function
	- [x] with unit tests
- [x] Add `groupData()` function
	- [x] doc
	- [x] with unit tests
- [x] We don't need an `update` method actually since simulation starts with separate mono and combo objects
- [x] Add `plot` method - also skipped for now. Both could be revisited later once we really go to use this in practice

### PR DESCRIPTION
**Pull request by danielinteractive**
closes #664

### CODE REVIEW

File: tests/testthat/helper-data.R
Comment (Puzzled-Face) [inline tests/testthat/helper-data.R:1]: A thought: The new `.DefaultXXXX()` constructors could probably replace the utility functions in the future.

File: R/Data-class.R
Comment (Puzzled-Face) [inline R/Data-class.R:1]: Please add a `.DefaultDataGrouped` constructor.

**Review by Puzzled-Face — CHANGES_REQUESTED** [review]

File: R/Data-class.R
Comment (danielinteractive) [reply R/Data-class.R:1]: thx, done

File: tests/testthat/helper-data.R
Comment (danielinteractive) [reply tests/testthat/helper-data.R:1]: maybe, but typically the default constructors produce pretty "empty" objects while for tests we typically need more "realistic" objects

**Review by Puzzled-Face — APPROVED** [review]

**Pull request merged**